### ISSUE TITLE
@ember-data/debug/data-adapter.js: Cannot read properties of undefined (reading 'env')

### ISSUE THREAD

**Issue by Jopie01** [emberjs/data#9461]
I wanted to upgrade ember-data from `5.4.0-alpha.57` to `5.4.0-alpha.73` but walked into trouble. So I decided to see how far I could go in the releases and ended up with `5.4.0-alpha.64`. Knowing from where to start I cloned the repo, linked it to my app and did a commit by commit test in my app and I ended up until PR #9292. After that commit I got the error:

```
Build Error (Babel) in @ember-data/debug/data-adapter.js

/home/user/ember/trytonstore/@ember-data/debug/data-adapter.js: Cannot read properties of undefined (reading 'env')TypeError: /home/user/ember/trytonstore/@ember-data/debug/data-adapter.js: Cannot read properties of undefined (reading 'env')
```

I suppose, because this is new way of building the packages, I have to make some changes as well in my app?
I'm using:
```
ember-source: 5.8.0
ember-cli: 5.8.0
@ember-data/debug: *
@ember-data/graph: *
@ember-data/json-api: *
@ember-data/request: *
@ember-data/store: *
@ember-data/tracking: *
```

**Comment by runspired** [comment]
on current canary you need to install `@warp-drive/build-config` and setConfig within ember-cli-build (see test apps for examples). This is something we're working to find a way to avoid being a breaking change before shipping to stable, but the old style of config will be deprecated and this new style is going to be a required thing for apps to do.

**Comment by Jopie01** [comment]
Thanks for explanation! I installed `@warp-drive/build-config` and changed the `ember-cli-build.js` and now everything is building successfully again. Now I have to dig through a lot of changes because of the changes in the Schema service.

FTR, I have installed the `5.4.0-alpha.73` versions of the different packages and changed the `ember-cli-build.js` as follows

```javascript
module.exports = async function (defaults) {
  const { setConfig } = await import('@warp-drive/build-config');

  .....

  setConfig(app, __dirname, {
    compatWith: process.env.EMBER_DATA_FULL_COMPAT ? '99.0' : null,
  });

  ....

};
```

Mind the `async` and `await`.
